camper_program.py

```diff
@@ -3,105 +3,394 @@
+    # and omit the 'self' parameter that instance methods would receive:
+
+    @staticmethod
+    def squares(num):
+        return num ** 2
+
+    @staticmethod
+    def square_roots(num):
+        return math.sqrt(num)
+
+    @staticmethod
+    def any_power(num, exp):
+        return pow(num, exp)
+
+    @staticmethod
+    def evalFraction(fracstr):
+        if "/" in fracstr:
+            spl = fracstr.split("/")
+            return float(spl[0]) / float(spl[1])
+        else:
+            return fracstr
+
+    @staticmethod
+    def isFrac(fracstr):
+        if "/" in fracstr:
+            return True
+        else:
+            return False
+
+
+class CalculatorREPL:
+    def __init__(self, calculator=None):
+        # Parametrize so that we can specify
+        # a different calculator implementation:
+        self.calc = Calculator if calculator is None else calculator
+
+    def squares(self):
+        print("Choose a number 1 through 100.")
+        number = int(input())
+        square = self.calc.squares(number)
+        print("The square of", number, "is", square)
+
+    def square_roots(self):
+        print("Choose a number 1 through 100.")
+        num = int(input())
+        root = self.calc.square_roots(num)
+        print("The square root of", num, "is", root)
+
+    def any_power(self):
+        num = float(input("Choose a number > "))
+        exp = input("Raise to what power? > ")
+        if "/" in exp:
+            exp = float(Fraction(exp))
+        else:
+            exp = float(exp)
+        print(self.calc.any_power(num, exp))
+
+    def isFrac(self, f):
+        isfrac = self.calc.isFrac(f)
+        if isfrac:
+            print(f + " is a fraction")
+        else:
+            print(f + " is not a fraction")
+        return isfrac
+
+    def get_decimal(self):
+        print("We are here")
+        The_answer = input(
+            'Would you like to find the decimal value of a fraction type "1" >>> '
+        )
+        if The_answer != "1":
+            return
+        print("Type a fraction like _/_ ")
+        validfrac = False
+        while not validfrac:
+            fraction = input("What is your fraction >>> ")
+            validfrac = self.isFrac(fraction)
+        print("Your fraction = ", self.calc.evalFraction(fraction))
+
+    def get_menu_options(self):
+        # this could come in handy
+        return {
+            "1": ["Squares", self.squares],
+            "2": ["Square roots", self.square_roots],
+            "3": ["Raise to any power", self.any_power],
+            "4": ["Get decimal", self.get_decimal],
+            "0": ["Exit", None],
+        }
+
+    def run_loop(self):
+        menu_options = self.get_menu_options()
+        looping = True
+        while looping:
+            for n, (menustr, _) in menu_options.items():
+                print(f"            {n}. {menustr}")
+            print("\n            Pick one please")
+
+            try:
+                do_it = input("    >>> ")
```

Comment:
I did not change this variable name.

https://en.wikipedia.org/wiki/Variable_(computer_science)#Naming_conventions

[![Starsky & Hutch (3/5) Movie CLIP - Do It (2004) HD](https://img.youtube.com/vi/P9pa_8-WdlU/0.jpg)](https://www.youtube.com/watch?v=P9pa_8-WdlU)